swap/src/api/request.rs

```diff
@@ -0,0 +1,667 @@
+                    amount,
+                );
+
+                tokio::select! {
+                    result = event_loop => {
+                        result
+                            .context("EventLoop panicked")?;
+                    },
+                    result = bob::run(swap) => {
+                        result
+                            .context("Failed to complete swap")?;
+                    }
+                }
+                json!({
+                    "empty": "true"
+                })
+            }
+            Method::History => {
+                let swaps = context.db.all().await?;
+                let mut vec: Vec<(Uuid, String)> = Vec::new();
+                for (swap_id, state) in swaps {
+                    let state: BobState = state.try_into()?;
+                    vec.push((swap_id, state.to_string()));
+                }
+
+                json!({ "swaps": vec })
+            }
+            Method::RawHistory => {
+                let raw_history = context.db.raw_all().await?;
+                json!({ "raw_history": raw_history })
+            }
+            Method::GetSeller => {
+                let swap_id = self.params.swap_id.context("Parameter swap_id is needed")?;
+                let peerId = context
+                    .db
+                    .get_peer_id(swap_id)
+                    .await
+                    .with_context(|| "Could not get PeerID")?;
+
+                let addresses = context
+                    .db
+                    .get_addresses(peerId)
+                    .await
+                    .with_context(|| "Could not get addressess")?;
+
+                json!({
+                    "peerId": peerId.to_base58(),
+                    "addresses": addresses
+                })
+            }
+            Method::SwapStartDate => {
+                let swap_id = self
+                    .params
+                    .swap_id
+                    .context("Parameter swap_id is missing")?;
+
+                let start_date = context.db.get_swap_start_date(swap_id).await?;
+
+                json!({
+                    "start_date": start_date,
+                })
+            }
+            Method::Config => {
+                let data_dir_display = context.config.data_dir.display();
+                tracing::info!(path=%data_dir_display, "Data directory");
+                tracing::info!(path=%format!("{}/logs", data_dir_display), "Log files directory");
+                tracing::info!(path=%format!("{}/sqlite", data_dir_display), "Sqlite file location");
+                tracing::info!(path=%format!("{}/seed.pem", data_dir_display), "Seed file location");
+                tracing::info!(path=%format!("{}/monero", data_dir_display), "Monero-wallet-rpc directory");
+                tracing::info!(path=%format!("{}/wallet", data_dir_display), "Internal bitcoin wallet directory");
+
+                json!({
+                    "result": []
```

Comment:
We need to return the file locations as JSON here but you've already mentioned this in the PR description.